### ISSUE TITLE
Windows Server 2012 Agents Disconnecting

### ISSUE THREAD

**Issue by asasin114** [Ylianst/MeshCentral#1656]
Have a bit of a strange one at the moment. All of my Windows Server 2012 Servers are showing that they are offline but I see them connect for a second or 2 every once in a while. When I look at the Power State it shows that everything is working just fine but it is showing offline in Mesh. I don't have any weird errors or anything and I'm running 0.5.95

**Comment by rowds15** [comment]
yes i have the same issue Version 0.5.95 - Uninstall and reinstall seems to fix the issue

entry in client MeshAgent.log file
[2020-07-28 07:50:10 AM] FATAL EXCEPTION [MeshAgent_XXXXXXXXXXXXXXXX.exe] @ [FuncAddr: 0x00007ff7d9419068 / BaseAddr: 0x00007ff7d92eb6f4 / Delta: -1235316]

**Comment by asasin114** [comment]
@rowds15 Uninstall and reinstall the agent?

**Comment by krayon007** [comment]
> yes i have the same issue Version 0.5.95 - Uninstall and reinstall seems to fix the issue

That's wierd... What if you just restarted the agent?

@asasin114 , do you have a log indicating a crash too?

**Comment by asasin114** [comment]
@krayon007 Sure do

[2020-01-11 01:25:09 PM] FATAL EXCEPTION [MeshAgent_E7BC4772A5B427C7.exe] @ [FuncAddr: 0x00007ff7a9b0507e / BaseAddr: 0x00007ff7a9b5b6ac / Delta: 353838]

**Comment by asasin114** [comment]
Can confirm that running the agent application and hitting the Install/Update button makes it work again just fine...

**Comment by rowds15** [comment]
> 
> 
> @rowds15 Uninstall and reinstall the agent?

yes

**Comment by Ylianst** [comment]
Looking into this crash status now.

**Comment by Ylianst** [comment]
This is what I get looking at fdelta on the 64bit MeshAgent of v0.5.95.

```
C:\Temp\node_modules\meshcentral\agents>meshservice64 -fdelta -1235316
[BaseAddr: 0x00007ff740c4b6f4, __chkstk]
```

Oddly the agent hash for this version is:

```
51930CE2658BCF5D5CC5FA60AACE3B85FBAB009BFDB3156EF12065F0BB7F2A58AF635EBAD0B486D9D559439F3BAB44F8
```

That does not match the hash in the exe name, I will have to figure this out later. Looking to see if I have any core dumps that would match this.

**Comment by Ylianst** [comment]
It does not seem like I have a dumpfile that matches the error above.

**Comment by Ylianst** [comment]
Quick update on this. Bryan has root causes the issue and has a proposed fix for it. Still needs more testing. Thank you for sending crash dumps, it was super useful.

**Comment by Ylianst** [comment]
Just published MeshCentral v0.5.97 with Bryan's fix for Windows Server 2012. Let us know if it works.
